### ISSUE TITLE
Feature: Most Favored Contract Status

### ISSUE THREAD

**Issue by sambacha** [foundry-rs/foundry#1046]
### Component

Forge

### Describe the feature you would like

When I run:
```shell
$ gitinit
# Initialized empty Git repository in $(pwd)/.git/
# [master (root-commit) 000e0e3]
forge install OpenZeppelin/openzeppelin-contracts
# Installing openzeppelin-contracts in "$(pwd)/lib/openzeppelin-contracts", (url: https://github.com/OpenZeppelin/openzeppelin-contracts, tag: None)
#    Installed openzeppelin-contracts
forge config > foundry.toml
```
generates a foundry.toml file with the remappings set as
```toml
remappings = ['openzeppelin-contracts/=lib/openzeppelin-contracts/']
```
but I want them to be
```toml
# note, we have to add `/contracts` to the remapping as well as the `@scope`
remappings = ['@openzeppelin/contracts/=lib/openzeppelin-contracts/contracts']
```

TLDR: remappings should ideally take the import statement format of nodejs 

### Additional context

Also, what would be nice is if foundry had a list of often used contracts and enforced this style of install:

```shell
tag=$(git describe --tags `git rev-list --tags --max-count=1`) && git checkout $tag -b $tag
```

Something similar to what @mds1 was talking about vis a vie install via git tag

The git history currently looks like this
```shell
* 76fca3ae - (HEAD -> master, origin/master, origin/HEAD) Add ERC721 and ERC1155 receiver support in Governor, Timelock (#3230) (28 hours ago) <Ashwin Yardi>
```
This is how it would look like if we used the git tag + checkout tag method above
```shell
* a5445b0a - (HEAD -> latest, tag: v4.5.0) 4.5.0 (6 weeks ago) <Francisco Giordano>
```

**Comment by sambacha** [comment]
note that trying to run `forge install @openzeppelin/contracts` attempts to install https://github.com/openzeppelin/contracts which is non-existent

**Comment by sambacha** [comment]
## OpenZeppelin Statistics 

 > sourced from npm directly 

| OpenZeppelin Version    | Download Hits |
|-------------------------|---------------|
| 3.4.1-solc-0.7-2        | "17,987"      |
| 4.2.0                   | "2,217"       |
| 4.3.2                   | "6,800"       |
| 4.3.3                   | "3,734"       |
| 4.4.2                   | "10,739"      |
| 4.5.0                   | "51,497"      |
| 3.4.1                   | "2,654"       |
| 2.5.1                   | "2,678"       |
|                         |               |
|                         |               |
| @openzeppelin/contracts | "4,831,188"   |
| 2021                    | 2732216       |
| 2022-3months            | 1466502       |
| 2022 estimated          | 5866008       |
| change y/oy/            | 3133792       |

This shows the most popular versions of openzeppelin and their download hits as reported by npm.com. 

For comparison, openzeppelin was downloaded about  ~2,732,216 times all of last year.

This year they are on track to do ~ 5,866,008 downloads.

**Comment by sambacha** [comment]
## Solidity Versions and Feature Sets

> Narrowing down compiler versions 

| **Features**             	| **Supported**    	| **Not Available **                               	|
|---------------------------|------------------	|-----------------------------------------------|
| receive-keyword          	| =0.6.0: true    	| <0.6.0: false                                  	|
| fallback-keyword         	| =0.6.0: true    	| <0.6.0: false                                  	|
| array-parameter-location 	| =0.7.0: memory  	| ^0.5.0 \|\| ^0.6.0: calldata<0.5.0: undefined 	|
| abiencoder-v2            	| =0.8.0: default 	| <0.8.0: experimental                             	|
| global-structs           	| =0.6.0: true    	| <0.6.0: false                                   	|
| structs-in-interfaces    	| =0.5.0: true    	| <0.5.0: false                                  	|
| custom-errors            	| =0.8.4: true    	| <0.8.4: false                                  	|
| user-defined-value-types 	| =0.8.8: true     	| <0.8.8: false                                  	|

**Comment by onbjerg** [comment]
I'm really unsure what you're asking for here given the amount of (seemingly) unrelated information that has been dumped. Do you want `forge install` to resolve npm packages?

**Comment by sambacha** [comment]
> I'm really unsure what you're asking for here given the amount of (seemingly) unrelated information that has been dumped. Do you want `forge install` to resolve npm packages?

I want to find the subset of compiler versions and openzeppelin versions that cover the maximal amount of user base. Eg what versions cover 80% of users for install OpenZeppelin and which are the corresponding solidity compiler versions needed to cover those versions. Knowing that it can be included in a container image so that it reduces time for getting started etc. i should have been more explicit yes

**Comment by onbjerg** [comment]
All of those solc versions are already supported, so I'm still not quite sure what the ask is :thinking:

**Comment by sambacha** [comment]
> All of those solc versions are already supported, so I'm still not quite sure what the ask is 🤔

This would be more appropriate as a discussion thread than an issue.  Thanks for the comments has helped reasoning about this better (ie what assortment of included compilers yields the greatest coverage to eliminate fetching them at invocation) 

the answer is the latest for every version except the latest semver which is latest and half of latest
